packages/edit-site/src/store/actions.js

```diff
@@ -676,3 +676,20 @@ export const toggleDistractionFree =
 				);
 		} );
 	};
+
+/**
+ * Sets the type of page content focus. Can be one of:
+ *
+ * - `'disableTemplate'`: Disable the blocks belonging to the page's template.
+ * - `'hideTemplate'`: Hide the blocks belonging to the page's template.
+ *
+ * @param {'disbleTemplate'|'hideTemplate'} pageContentFocusType The type of page content focus.
+ *
+ * @return {Object} Action object.
+ */
+export function setPageContentFocusType( pageContentFocusType ) {
+	return {
+		type: 'SET_PAGE_CONTENT_FOCUS_TYPE',
+		pageContentFocusType,
+	};
+}
```

Comment:
This might be getting a little nit-picky, so my apologies if this isn't useful feedback! Begin thoughts:

Just looking over this again, and I'm wondering a bit about the addition of `setPageContentFocusType` and `getPageContentFocusType` and the potential overlap between these and `setHasPageContentFocus`. Since the latter is already a non-private action, I'm not sure how well we can consolidate things, but I'm wondering if it might be possible?

My main concern is that `setPageContentFocusType`'s state is pretty much dependent on `setPageContentFocus` being `true`, so depending on the circumstances it might be awkward having both bits of state.

Just thinking out loud, but would it be possible to consolidate via something like:

* Update `hasPageContentFocus` selector and reducer to look at the same piece of state as `getPageContentFocusType`, but the `hasPageContentFocus` just checks that the value is truthy.
* Update `setPageContentFocus` so that it's string-based, but if it accepts a `boolean` then `true` maps to `disableTemplate`?

I think that could group things together, though it might not be desirable if we ever wish to persist the state of the focus type separately to whether or not page content focus is set. If that's the case then feel free to disregard this feedback!

On the other hand, if we're undecided about which direction we'd like to go in the future, then another option could be to move the action and selector added in this PR to be private.

Reply:
With @noisysocks gone it's hard to gauge the intention behind `pageContentFocusType` — maybe other "types" were envisioned? 

I agree it seems to be pretty tightly coupled with `hasPageContentFocus`.

Thanks for thinking about an alternative. I think we could take a different approach as you mentioned, but I'm not convinced that expanding the return type of an already public selector with a bunch of usages will make things easier to work with in the future.

Also, as you mention there could be a case in which the states are independent. Locking a template while editing a page vs hiding a template while editing a page seem to me to be two separate actions.

> another option could be to move the action and selector added in this PR to be private

I was actually in the middle of doing this, so I think our instincts are in alignment insofar as the ambiguity of the selector 😄

Reply:
> I'm not convinced that expanding the return type of an already public selector with a bunch of usages will make things easier to work with in the future.

Very good point! Best not to mess with the public signature 👍

> I was actually in the middle of doing this, so I think our instincts are in alignment insofar as the ambiguity of the selector 😄

Nice one! Sounds good 🙂